services/provider/proto/provider.proto

```diff
@@ -143,6 +143,8 @@ message GetDesiredClientStateResponse{
     // kubeObjects holds the definition and metadata for all kube objects to be reconciled
     // by the client operator
     repeated KubeObject kubeObjects = 6;
+    // NonResilientPoolsTopologyKey for replica-1
+    string nonResilientPoolsTopologyKey = 7;
```

Comment:
When it comes to the GRPC interface I would like the interface to be more generalised for future needs. There are a couple of questions the need to be answered:
- Are we always going to support a single a single topological key?
- Are there cases where we need topological keys that are not related to non Resilient pools?